Celeste.Mod.mm/Patches/EntityData.cs

```diff
@@ -12,5 +12,19 @@ class patch_EntityData : EntityData {
             return orig_Has(key);
         }
 
+        /// <summary>
+        /// Get the <see cref="T:System.String" /> value associated with a key.
+        /// An Empty or pure Whitespace value will result in the <paramref name="defaultValue"/> to be returned.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="defaultValue"></param>
+        /// <returns></returns>
+        public string String(string key, string defaultValue = null) {
+            if (Values != null && Values.TryGetValue(key, out object value)) {
+                if (!string.IsNullOrWhiteSpace((string) value))
```

Comment:
if `value` is not string, it would crash

Reply:
I'll add a type pattern to see if it's a string, would that work?

Reply:
you need tostring i guess

Reply:
so `!string-IsNullOrWhiteSpace(value.ToString())` ?

Reply:
Or is there a way to do pattern matching